applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_high_cycle_fatigue_law.cpp

```diff
@@ -0,0 +1,476 @@
+void GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::SetValue(
+    const Variable<double>& rThisVariable,
+    const double& rValue,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    if (rThisVariable == DAMAGE || rThisVariable == UNIAXIAL_STRESS || rThisVariable == THRESHOLD) {
+        GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::SetValue(rThisVariable, rValue, rCurrentProcessInfo);
+    } else if (rThisVariable == FATIGUE_REDUCTION_FACTOR) {
+        mFatigueReductionFactor = rValue;
+    } else if (rThisVariable == NUMBER_OF_CYCLES) {
+        mNumberOfCycles = rValue;
+    } else {
+        return GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::SetValue(rThisVariable, rValue, rCurrentProcessInfo);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+double& GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::GetValue(
+    const Variable<double>& rThisVariable,
+    double& rValue
+    )
+{
+    if (rThisVariable == DAMAGE || rThisVariable == UNIAXIAL_STRESS || rThisVariable == THRESHOLD) {
+        GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::GetValue(rThisVariable, rValue);
+    } else if (rThisVariable == FATIGUE_REDUCTION_FACTOR) {
+        rValue = mFatigueReductionFactor;
+    } else if (rThisVariable == NUMBER_OF_CYCLES) {
+        rValue = mNumberOfCycles;
+    } else {
+        return GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::GetValue(rThisVariable, rValue);
+    }
+    return rValue;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+Matrix& GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Matrix>& rThisVariable,
+    Matrix& rValue
+    )
+{
+    if (rThisVariable == INTEGRATED_STRESS_TENSOR) {
+        rValue = MathUtils<double>::StressVectorToTensor(this->GetStressVector());
+    } else if (rThisVariable == CONSTITUTIVE_MATRIX) {
+        this->CalculateElasticMatrix(rValue, rParameterValues);
+    }
+    return rValue;
+}
+
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template <class TConstLawIntegratorType>
+Vector& GenericSmallStrainHighCycleFatigueLaw<TConstLawIntegratorType>::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Vector>& rThisVariable,
+    Vector& rValue
+    )
+{
+    return GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::CalculateValue(rParameterValues, rThisVariable, rValue);
+}
```

Comment:
same, please remove if it only calls the same function in the baseclass

Reply:
done